std-bits/table/src/main/java/org/enso/table/data/table/Table.java

```diff
@@ -43,6 +43,11 @@ public Table(Column[] columns, AggregatedProblems problems) {
       throw new IllegalArgumentException("A Table must have at least one column.");
     }
 
+    boolean uniqueColumns = Arrays.stream(columns).map(Column::getName).distinct().count() == columns.length;
+    if (!uniqueColumns) {
+      throw new IllegalArgumentException("Column names must be unique within a Table.");
+    }
```

Comment:
Worth putting a test in for this I think - appreciate we should never have it but worth the safety check it is never removed.

Reply:
Good idea